Ruby 4.1 (ruby-head) Support

# Add Ruby 4.1 version support for ruby-head compatibility

## Description

When building gems that depend on `rb-sys` against Ruby head (4.1.0), the version comparison cfg flags for Ruby 4.1 are not available. This prevents dependent crates like `magnus` from using conditional compilation to handle API changes introduced in Ruby 4.1.

## Problem

Ruby 4.1 (currently ruby-head as of January 2026) introduced a breaking change in the `rb_data_type_struct` C struct:
- The `reserved` field was removed
- A new `handle_weak_references` field was added

Crates like [magnus](https://github.com/matsadler/magnus/) need to conditionally compile different code paths based on Ruby version:

```rust
#[cfg(ruby_lt_4_1)]
function: rb_data_type_struct__bindgen_ty_1 {
    // ... 
    reserved: [ptr::null_mut(); 1],
},
#[cfg(ruby_gte_4_1)]
function: rb_data_type_struct__bindgen_ty_1 {
    // ...
    handle_weak_references: None,
},
```

However, the cfg flags `ruby_lt_4_1`, `ruby_gte_4_1`, etc. are not emitted because Ruby 4.1 is not in the `SUPPORTED_RUBY_VERSIONS` array.

### Error Message

When building with Ruby head without this fix:

```
error[E0560]: struct `rb_data_type_struct__bindgen_ty_1` has no field named `reserved`
   --> magnus-0.8.2/src/typed_data.rs:352:17
    |
352 |                 reserved: [ptr::null_mut(); 1],
    |                 ^^^^^^^^ `rb_data_type_struct__bindgen_ty_1` does not have this field
    |
    = note: available fields are: `handle_weak_references`
```

### Affected Files

1. `crates/rb-sys/build/main.rs` - `SUPPORTED_RUBY_VERSIONS` array
2. `crates/rb-sys-env/src/ruby_version.rs` - `COMPARABLE_RUBY_MINORS` array

## Impact

Gems that rely on magnus, and thus rb_sys, are unable to run CI against ruby-head.

--- a/crates/rb-sys-env/src/ruby_version.rs
+++ b/crates/rb-sys-env/src/ruby_version.rs
@@ -2,7 +2,7 @@ use std::collections::HashMap;
 
 const COMPARABLE_RUBY_MAJORS: [u8; 4] = [1, 2, 3, 4];
 
-const COMPARABLE_RUBY_MINORS: [(u8, u8); 11] = [
+const COMPARABLE_RUBY_MINORS: [(u8, u8); 13] = [
     (2, 2),
     (2, 3),
     (2, 4),
@@ -14,6 +14,8 @@ const COMPARABLE_RUBY_MINORS: [(u8, u8); 11] = [
     (3, 2),
     (3, 3),
     (3, 4),
+    (4, 0),
+    (4, 1),
 ];
 
 /// The current Ruby version.

--- a/crates/rb-sys/build/main.rs
+++ b/crates/rb-sys/build/main.rs
@@ -13,7 +13,7 @@ use std::{
 };
 use version::Version;
 
-const SUPPORTED_RUBY_VERSIONS: [Version; 11] = [
+const SUPPORTED_RUBY_VERSIONS: [Version; 12] = [
     Version::new(2, 3),
     Version::new(2, 4),
     Version::new(2, 5),
@@ -25,6 +25,7 @@ const SUPPORTED_RUBY_VERSIONS: [Version; 11] = [
     Version::new(3, 3),
     Version::new(3, 4),
     Version::new(4, 0),
+    Version::new(4, 1),
 ];
 
 fn main() {

